packages/applicant/src/pages/api/create-mandatory-question.page.tsx

```diff
@@ -11,17 +11,36 @@ export default async function handler(
   try {
     const grantMandatoryQuestionService =
       GrantMandatoryQuestionService.getInstance();
-    const mandatoryQuestionId =
+    const mandatoryQuestion =
       await grantMandatoryQuestionService.createMandatoryQuestion(
         schemeId,
         getJwtFromCookies(req)
       );
 
-    res.redirect(
-      `${process.env.HOST}${routes.mandatoryQuestions.namePage(
-        mandatoryQuestionId
-      )}`
-    );
+    const organisationProfileKeys = [
+      'name',
+      'addressLine1',
+      'city',
+      'postcode',
+      'orgType',
+      'companiesHouseNumber',
+      'charityCommissionNumber',
+    ];
+
+    const areOrganisationProfileQuestionsComplete =
+      organisationProfileKeys.every(
+        (key) => mandatoryQuestion[key] !== undefined
+      ) &&
+      organisationProfileKeys.every((key) => mandatoryQuestion[key] !== null) &&
+      organisationProfileKeys
+        .filter((key) => key !== 'orgType')
+        .every((key) => mandatoryQuestion[key] !== '');
+
+    const redirectionUrl = areOrganisationProfileQuestionsComplete
+      ? routes.mandatoryQuestions.fundingAmountPage(mandatoryQuestion.id)
+      : routes.mandatoryQuestions.namePage(mandatoryQuestion.id);
+
+    return res.redirect(`${process.env.HOST}${redirectionUrl}`);
```

Comment:
This is very inefficient your making 3 loops, consider this instead

```
const areOrganisationProfileQuestionsComplete = organisationProfileKeys.every((key) => {
  const value = mandatoryQuestion[key];

  if (key === 'orgType') {
    return value !== null && value !== undefined;
  }

  return value !== '' && value !== null && value !== undefined;
});
```

Reply:
absolutely right... i also forgot that in the fe, orgType is a string, so there's no need to differentiate